src/components/DirectoryContentTable.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { defaultColumnDefinition } from './utils/directory-content-utils';
+import { CustomAGGrid, ElementType } from '@gridsuite/commons-ui';
+import { AgGridReact } from 'ag-grid-react';
+import { IElement } from '../redux/reducer.type';
+import { GetRowIdParams } from 'ag-grid-community/dist/types/core/interfaces/iCallbackParams';
+import { ColDef, GridReadyEvent, RowClassParams } from 'ag-grid-community';
+
+interface DirectoryContentTableProps {
+    gridRef: React.ForwardedRef<AgGridReact<IElement>>;
+    rows: IElement[];
+    handleRowSelected: () => void;
+    handleCellClick: () => void;
+    colDef: ColDef[];
+}
+
+const onGridReady = ({ api }: GridReadyEvent<IElement>) => {
+    api?.sizeColumnsToFit();
+};
+
+const getRowId = (params: GetRowIdParams<IElement>): string =>
```

Comment:
```suggestion
const getRowId = (params: GetRowIdParams<IElement>) =>
```
No need to type return type when it can be inferred
Here you are even changing "UUID" to "string" type by doing so

Reply:
Done